lib/render/RayCaster.cpp

```diff
@@ -562,46 +524,55 @@ int RayCaster::_initializeGL()
 
 int RayCaster::_paintGL( bool fast ) 
 {
+    // Collect params and grid that will be used repeatedly
+    RayCasterParams* params = dynamic_cast<RayCasterParams*>( GetActiveParams() );
+    if( !params )
+    {
+        MyBase::SetErrMsg("Error occured during retrieving RayCaster parameters!");
+        glBindVertexArray( 0 );
+        glBindBuffer( GL_ELEMENT_ARRAY_BUFFER, 0 );
+        return PARAMSERROR;
+    }
+    // Do not perform any fast rendering in cell traverse mode
+    int castingMode  =  int(params->GetCastingMode());
+    if( castingMode == CellTraversal && fast )
+        return 0;
+
+    StructuredGrid*  grid = nullptr;
+    if( _userCoordinates.GetCurrentGrid( params, _dataMgr, &grid ) != 0 )
+    {
+        MyBase::SetErrMsg( "Failed to retrieve a StructuredGrid" );
+        glBindVertexArray( 0 );
+        glBindBuffer( GL_ELEMENT_ARRAY_BUFFER, 0 );
+        return GRIDERROR;
+    }
+
+#ifndef NDEBUG
+    // Do NOT try to reload shaders in Release mode due to 
```

Comment:
The ShaderManager does this automatically.

Reply:
This macro is removed by commit 4d66441